core/src/stack/outputs.rs

```diff
@@ -219,21 +219,21 @@ impl ToElements<Felt> for StackOutputs {
 impl Serializable for StackOutputs {
     fn write_into<W: ByteWriter>(&self, target: &mut W) {
         debug_assert!(self.stack.len() <= u32::MAX as usize);
-        target.write_u32(self.stack.len() as u32);
+        target.write_usize(self.stack.len());
         target.write_many(&self.stack);
 
         debug_assert!(self.overflow_addrs.len() <= u32::MAX as usize);
-        target.write_u32(self.overflow_addrs.len() as u32);
+        target.write_usize(self.overflow_addrs.len());
         target.write_many(&self.overflow_addrs);
     }
 }
 
 impl Deserializable for StackOutputs {
     fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
-        let count = source.read_u32()?.try_into().expect("u32 must fit in a usize");
+        let count = source.read_usize()?;
         let stack = source.read_many::<u64>(count)?;
 
-        let count = source.read_u32()?.try_into().expect("u32 must fit in a usize");
+        let count = source.read_usize()?;
         let overflow_addrs = source.read_many::<u64>(count)?;
```

Comment:
We should make sure the counts are smaller than `u16:MAX` and return an error if not.